name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComputersharePDFExtractor.java

```diff
@@ -0,0 +1,172 @@
+
+/**
+ * Extractor for Computershare (https://www.computershare.com/) depot documents.
+ * Supported: * Import transaction purchase entries from Statement documents
+ * (only document CS provides)
+ */
+@SuppressWarnings("nls")
+public class ComputersharePDFExtractor extends AbstractPDFExtractor
+{
+
+    private static final String BANK_IDENTIFIER = "Computershare";
+    private static final String BANK_LABEL = "Computershare Trust Company, N.A.";
+    private static final String BLOCK_IDENTIFIER_PURCHASE = "^[\\d]{2} [\\w]{3} [\\d]{4} Purchase.*";
+    private static final String DOCUMENT_IDENTIFIER_STATEMENT_DOC = ".*Plan Statement";
+    private static final String PATTERN_KEY_AMOUNT = "amount";
+    private static final String PATTERN_KEY_CURRENCY = "currency";
+    private static final String PATTERN_KEY_DATE = "date";
+    private static final String PATTERN_KEY_FEE = "fee";
+    private static final String PATTERN_KEY_NAME = "name";
+    private static final String PATTERN_KEY_SHARES = "shares";
+    private static final String PATTERN_KEY_TICKERSYMBOL = "tickerSymbol";
+    private static final String PATTERN_KEY_WKN = "wkn";
+    private static final String PATTERN_PURCHASE_ENTRY = "^(?<date>[\\d]{2} [\\w]{3} [\\d]{4}) Purchase (?<amount>[\\.,\\d]+) (?<fee>([\\.,\\d]+)\\s)?(?<netAmount>[\\.,\\d]+) (?<grantDate>[\\d]{2} [\\w]{3} [\\d]{4}) (?<fmvGrant>[\\.,\\d]+) (?<purchaseDate>[\\d]{2} [\\w]{3} [\\d]{4}) (?<fmvPurchase>[\\.,\\d]+) (?<sharePrice>[\\.,\\d]+) (?<shares>[\\.,\\d]+) (?<totalShares>[\\.,\\d]+).*";
+    private static final String PATTERN_STRING_CUSIP = "^(?<wkn>[A-Z0-9]{9})$";
+    private static final String PATTERN_STRING_DOUBLE_DATE = "^[\\d]{2} [\\w]{3} [\\d]{4} [\\d]{2} [\\w]{3} [\\d]{4}$";
+    private static final String PATTERN_STRING_SINGLE_DATE = "^[\\d]{2} [\\w]{3} [\\d]{4}$";
+    private static final String PATTERN_STRING_TICKERSYMBOL = "^(?<tickerSymbol>\\S*)$";
+
+    /**
+     * @param client
+     */
+    public ComputersharePDFExtractor(Client client)
+    {
+        super(client);
+
+        // Add bank identifier to be able identify the bank documents
+        addBankIdentifier(BANK_IDENTIFIER);
+        // register depot statement processing
+        addDepotStatementTransaction();
+    }
+
+    private void addDepotStatementTransaction()
+    {
+
+
+        final DocumentType type = new DocumentType(DOCUMENT_IDENTIFIER_STATEMENT_DOC, 
+                        (context, lines) -> {
+                            // currently no parsable security information
+                            // identifier in statement document - needs line
+                            // guessing
+                            final Pattern singleDatePattern = Pattern.compile(PATTERN_STRING_SINGLE_DATE);
+                            final Pattern doubleDatePattern = Pattern.compile(PATTERN_STRING_DOUBLE_DATE);
+                            final Pattern cusipPattern = Pattern.compile(PATTERN_STRING_CUSIP);
```

Comment:
I understand the thinking for extracting the pattern/attributes names into constants.
For reasons of readability, we have decided against it for most of the time.
Take the cusip pattern: the constant is defined and then used once, but now it is defined far away (in terms of lines) so that one cannot glance to see the context.

The thinking was: attributes must be consistent between ```match``` and ```assign``` but not beyond. So we scope it those methods.

Other than that it looks good to me (for now only looked in the browser)